php-checks/src/test/resources/checks/HardCodedSecretCheckAST.php

```diff
@@ -0,0 +1,259 @@
+}
+
+// Class Constants (as Variables in AST)
+class ContainsConstrants
+{
+  // Noncompliant@+1
+  const Token = "abcdefghijklmnopqrs";
+//              ^^^^^^^^^^^^^^^^^^^^^
+
+  const API_KEY = "abcdefghijklmnopqrs";  // Noncompliant
+  const AUTH = "abcdefghijklmnopqrs";               // Noncompliant
+
+  const OK_CONSTANT = "abcdefghijklmnopqrs";        // Compliant
+
+  // Assignments
+  function doSomething()
+  {
+    $someSecret = "";
+    $someSecret = "abcdefghijklmnopqrs";          // Noncompliant
+
+  }
+}
+
+// Class Properties
+class ContainsProperties
+{
+  public $passed = 'abcdefghijklmnopqrs';
+  public $apikey = 'abcdefghijklmnopqrs';                   // Noncompliant
+  public string $api_key = "abcdefghijklmnopqrs";           // Noncompliant
+
+  protected $credential = "abcdefghijklmnopqrs";            // Noncompliant
+  private $private_credential = "abcdefghijklmnopqrs";      // Noncompliant
+  static $static_credential = "abcdefghijklmnopqrs";        // Noncompliant
+
+  public function __construct(
+    public readonly string $auth = "abcdefghijklmnopqrs",             // Noncompliant
+    protected              $protected_auth = "abcdefghijklmnopqrs",   // Noncompliant
+    private                $private_auth = "abcdefghijklmnopqrs",     // Noncompliant
+                           $oauth = "abcdefghijklmnopqrs",            // Noncompliant
+                           $empty_auth
+  )
+  {
+    // empty
+  }
+
+}
+
+// Object
+$object = (object)[
+  'propertyOne' => 'abcdefghijklmnopqrs',
+  'secret' => 'abcdefghijklmnopqrs',        // Noncompliant
+  "token" => "abcdefghijklmnopqrs"          // Noncompliant
+];
+
+// Array
+$array = array(
+  'passed' => 'abcdefghijklmnopqrs',
+  'auth' => 'abcdefghijklmnopqrs',          // Noncompliant
+  "credential" => "abcdefghijklmnopqrs"     // Noncompliant
+);
+
+function compareStrings($auth)
+{
+  if ($auth == null)
+  {
+    return;
+  }
+  if ($auth == "")
+  {
+    return;
+  }
+  if ($auth == "X")
+  {
+    return;
+  }
+  if ($auth == "abcdefghijklmnopqrs")                     // Noncompliant
+  {
+    echo $auth;
+  }
+  if ("abcdefghijklmnopqrs" == $auth)                     // Noncompliant
+  {
+    echo $auth;
+  }
+  if ($auth === "abcdefghijklmnopqrs")                     // Noncompliant
+  {
+    echo $auth;
+  }
+  if ("abcdefghijklmnopqrs" === $auth)                     // Noncompliant
+  {
+    echo $auth;
+  }
+  if ($auth <=> "abcdefghijklmnopqrs")                     // Noncompliant
+  {
+    echo $auth;
+  }
+  if ("abcdefghijklmnopqrs" <=> $auth)                     // Noncompliant
+  {
+    echo $auth;
+  }
+  if (PASSED == $auth)                                    // FN unable to resolve the value
+  {
+    echo $auth;
+  }
+  if ($auth == PASSED)                                     // FN unable to resolve the value
+  {
+    echo $auth;
+  }
+
+  // case sensitive
+  if (strcmp($auth, "abcdefghijklmnopqrs"))         // Noncompliant
```

Comment:
Not sure if I saw test for `strcmp` with:
- no arguments
- only one arguments
- three arguments
Doesn't mean we have to support/do anything, more like do document/ensure we don't crash.